packages/components/src/Drawer/Drawer.component.js

```diff
@@ -100,27 +101,87 @@ export function cancelActionComponent(onCancelAction, getComponent) {
 	return (
 		<ActionComponent
 			{...enhancedCancelAction}
-			className={classnames(theme['tc-drawer-close-action'], enhancedCancelAction.className)}
+			className={classnames(
+				'tc-drawer-close-action',
+				theme['tc-drawer-close-action'],
+				enhancedCancelAction.className,
+			)}
 		/>
 	);
 }
 
-export function subtitleComponent(subtitle) {
+export function SubtitleComponent({ subtitle }) {
 	if (!subtitle || !subtitle.length) {
 		return null;
 	}
 	return <h2 title={subtitle}>{subtitle}</h2>;
 }
 
-function DrawerTitle({ title, subtitle, children, onCancelAction, getComponent }) {
+SubtitleComponent.propTypes = {
+	subtitle: PropTypes.string,
+};
+
+function DrawerTitle({
+	title,
+	subtitle,
+	children,
+	onCancelAction,
+	getComponent,
+	editable,
+	inProgress,
+	onEdit,
+	onSubmit,
+	actionCreatorSubmit,
+	onCancel,
+	dispatchActionCreator,
+	...props
+}) {
+	const [isEditMode, setIsEditMode] = React.useState(false);
+	function handleEdit(...args) {
+		setIsEditMode(true);
+		if (onEdit) {
+			onEdit(...args);
+		}
+	}
+	function handleCancel(...args) {
+		setIsEditMode(false);
+		if (onCancel) {
+			onCancel(...args);
+		}
+	}
+	function handleSubmit(...args) {
+		setIsEditMode(false);
+		if (typeof onSubmit === 'function') {
+			onSubmit(...args);
+		}
+		if (actionCreatorSubmit) {
+			dispatchActionCreator(actionCreatorSubmit, event, args[1]);
```

Comment:
you miss the check if `dispatchActionCreator`
This is very very CMF related and we are in components. 🤔
I would go to handle that in the onSubmit with the container.
Wait and see the opinion of others on this one

Reply:
thx, I bring it on project side